src/Extensions/Abstractions/ExceptionHelpers.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Helper methods for <see cref="Exception" />
+	/// </summary>
+	public static class ExceptionHelpers
+	{
+		/// <summary>
+		/// Returns true if exception could be handled or logged and false for unrecoverable exeptions
+		/// like <see cref="OutOfMemoryException" /> and <see cref="StackOverflowException" />
+		/// </summary>
+		public static bool IsRecoverable(this Exception exception) =>
+			!(exception is OutOfMemoryException)
+			|| !(exception is StackOverflowException);
```

Comment:
Starting with the .NET Framework 2.0, you can't catch a StackOverflowException object with a try/catch block, and the corresponding process is terminated by default (from https://docs.microsoft.com/en-us/dotnet/api/system.stackoverflowexception?redirectedfrom=MSDN&view=netframework-4.8)

Reply:
Also see my comment in another PR about OutOfMemory.